libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -1000,6 +1017,19 @@ protected virtual Task<ResourceResponse> ProcessOutgoingActivityAsync(ITurnConte
             throw new NotImplementedException();
         }
 
+        /// <summary>
+        /// Logic to build an <see cref="AppCreentials"/> object to be used to acquire tokens
+        /// for this HttpClient.
+        /// </summary>
+        /// <param name="appId">The application id.</param>
+        /// <param name="oAuthScope">The optional OAuth scope.</param>
+        /// <returns>The app credentials to be used to acquire tokens.</returns>
+        protected virtual async Task<AppCredentials> BuildCredentialsAsync(string appId, string oAuthScope = null)
+        {
+            var appPassword = await CredentialProvider.GetAppPasswordAsync(appId).ConfigureAwait(false);
+            return ChannelProvider != null && ChannelProvider.IsGovernment() ? new MicrosoftGovernmentAppCredentials(appId, appPassword, HttpClient, Logger) : new MicrosoftAppCredentials(appId, appPassword, HttpClient, Logger, oAuthScope);
```

Comment:
This line confuses me, as I can't figure out what it's doing. 

Can you break it down a bit and add some comments. For example, why does the OAuthScope only get passed into the App, but not Government, Credentials?

Reply:
Yes, this is a legacy line. I'll add comments for now but I don't like the way GovernmentCreds inherits from the password specific ones, we have some improvement plans for R8 here.